common/src/main/java/com/wynntils/core/features/Configurable.java

```diff
@@ -5,10 +5,34 @@
 package com.wynntils.core.features;
 
 import com.wynntils.core.config.ConfigHolder;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
-public interface Configurable {
-    void updateConfigOption(ConfigHolder configHolder);
+public abstract class Configurable {
```

Comment:
No, don't turn this into a class. We might want to make other things configurable.

Reply:
Make a common abstract base class `AbstractConfigurable` or whatever.

Reply:
I think its fine for now, as that's partially future thinking. Also, off the top of my head, I don't see a use case in which we need to extend configurable along with another class, but please prove me wrong.

Reply:
I disagree. There are design reasons to having an interface even if there is only a single implementation.

I want to hear @kristofbolyai's opinion on this as well. It's his design, after all.

At the very least, if you two agree on keeping this abstract class redesign, you must rename it. The `*able` naming is used in Java to signify an interface.

Reply:
I can't think of a use case where we would need to extend from this and some other class, right now. But that does not mean I think we should do this. I think this seems reasonable, but the only gain is that we can remove around 30 lines of code. I am not sure that is worth it.

Reply:
> Make a common abstract base class `AbstractConfigurable` or whatever.

Do this. This is what MC does with widgets for example. You have the widget interface, and `AbstractWidget` with some basic impl. This way, we get to keep the good of both worlds.